### PR TITLE
Feature/15 event callbacks

### PR DESCRIPTION
**Pull request by mjurczyk**
### Issue #15 

*What's new:*

* Added 6 new event callbacks (as requested in the issue)
* Moved submit action logic from layout generator method to the action method
* Added docs for new callbacks

### CODE REVIEW

File: src/pathfora.js [hunk 1]
```diff
@@ -320,6 +331,16 @@
       } else if (condition.showOnInit) {
         context.pathfora.showWidget(widget);
       }
+
+//    NOTE theoretically onInit should be here, @see comment on initializeWidgetArray
```
Comment (mjurczyk) [inline src/pathfora.js]: Discuss this.

**Comment by mjurczyk** [comment]
Discuss if the onModalClose callback should be called when the form is submitted (whether we want to track non-effective closes or all of them).

**Comment by araddon** [comment]
whoops, just commented on the ticket instead of here, but asked about global-level callbacks for any widget in the readme.

File: src/pathfora.js (hunk 1 above)
Comment (araddon) [inline src/pathfora.js]: thats fine

**Pull request merged**